src/Pages/Withdrawals.razor

```diff
@@ -355,59 +352,40 @@
     </Column>
 
 </Row>
-<PSBTSign @ref="_psbtSignRef" SignedPSBT="@_signedPSBT" RequestId="_selectedRequest?.Id" SigHashMode="SigHash.All"
-          TemplatePsbtString="@_templatePsbtString" ApproveRequestDelegate="ApproveRequestDelegate"></PSBTSign>
-<Modal @ref="_rejectCancelModalRef">
-
-    <ModalContent Centered>
-        <ModalHeader>
-            <ModalTitle>Wallet withdrawal: @_selectedRequest?.Id</ModalTitle>
-            <CloseButton />
-        </ModalHeader>
-        <ModalBody>
-            <Fields Flex="Flex.JustifyContent.Center">
-                <Field>
-                    <Validation @ref="_rejectCancelValidation" Validator="@(_rejectCancelStatus == WalletWithdrawalRequestStatus.Rejected ? ValidationRule.IsNotEmpty : ValidationRule.None)">
-                        <FieldLabel>Please type a reason before performing this operation</FieldLabel>
-                        <MemoEdit Rows="2" @bind-Text="@_cancelOrRejectReason">
-                            <ValidationError />
-                        </MemoEdit>
-                    </Validation>
-                </Field>
-            </Fields>
-        </ModalBody>
-        <ModalFooter>
-            <Button Color="Color.Secondary" Clicked="@HideRejectCancelModal">Cancel</Button>
-            <Button Disabled="@(_rejectCancelValidation?.Validate() == ValidationStatus.Error)" Color="Color.Primary" Clicked="RejectOrCancel">
-                Submit
-            </Button>
-        </ModalFooter>
-    </ModalContent>
-</Modal>
 
+<PSBTSign 
+    @ref="_psbtSignRef" 
+    SignedPSBT="@_signedPSBT" 
+    RequestId="_selectedRequest?.Id" 
+    SigHashMode="SigHash.All"
+    TemplatePsbtString="@_templatePsbtString" 
+    ApproveRequestDelegate="ApproveRequestDelegate"
+/>
+
+<CancelOrRejectPopup
+    @ref="@_rejectCancelModalRef"
+    Title='@("Wallet withdrawal:"+_selectedRequest?.Id)'
+    Reason="@_cancelOrRejectReason"
+    Validator="@RejectReasonValidator"
+    OnCancel="@HideRejectCancelModal"
+    OnSubmit="@RejectOrCancel"
+/>
 
-@code {
-    [Inject]
-    private IWalletWithdrawalRequestRepository WalletWithdrawalRequestRepository { get; set; }
-
-    [Inject]
-    private IWalletWithdrawalRequestPsbtRepository WalletWithdrawalRequestPsbtRepository { get; set; }
-
-    [Inject]
-    private IWalletRepository WalletRepository { get; set; }
-
-    [Inject]
-    private IBitcoinService BitcoinService { get; set; }
+@inject IToastService ToastService
+@inject IWalletWithdrawalRequestRepository WalletWithdrawalRequestRepository
+@inject IWalletWithdrawalRequestPsbtRepository WalletWithdrawalRequestPsbtRepository
+@inject IWalletRepository WalletRepository
+@inject IBitcoinService BitcoinService
+@inject ISchedulerFactory SchedulerFactory
```

Comment:
We can do injects in both ways, but it is true that we have not used the same approach always

Reply:
There were a lot of null warnings on this, that's why I moved it.
I tried to remove all null warnings but some of them are literally impossible to remove. For example when you have a ref in a component

Reply:
Yes null handling cannot be complete removed in C# even with modern features